ste/sourceInfoProvider-GCP.go

```diff
@@ -0,0 +1,167 @@
+package ste
+
+import (
+	"cloud.google.com/go/storage"
+	"fmt"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-azcopy/common"
+	"golang.org/x/oauth2/google"
+	"io/ioutil"
+	"net/url"
+	"time"
+)
+
+type gcpSourceInfoProvider struct {
+	jptm         IJobPartTransferMgr
+	transferInfo TransferInfo
+
+	rawSourceURL *url.URL
+
+	gcpClient   *storage.Client
+	gcpURLParts common.GCPURLParts
+}
+
+var gcpClientFactory = common.NewGCPClientFactory()
+var jsonKey []byte
+
+func newGCPSourceInfoProvider(jptm IJobPartTransferMgr) (ISourceInfoProvider, error) {
+	var err error
+	p := gcpSourceInfoProvider{jptm: jptm, transferInfo: jptm.Info()}
+
+	p.rawSourceURL, err = url.Parse(p.transferInfo.Source)
+	if err != nil {
+		return nil, err
+	}
+	p.gcpURLParts, err = common.NewGCPURLParts(*p.rawSourceURL)
+	if err != nil {
+		return nil, err
+	}
+
+	p.gcpClient, err = gcpClientFactory.GetGCPClient(
+		p.jptm.Context(),
+		common.CredentialInfo{
+			CredentialType:    common.ECredentialType.GoogleAppCredentials(),
+			GCPCredentialInfo: common.GCPCredentialInfo{},
+		},
+		common.CredentialOpOptions{
+			LogInfo:  func(str string) { p.jptm.Log(pipeline.LogInfo, str) },
+			LogError: func(str string) { p.jptm.Log(pipeline.LogError, str) },
+			Panic:    func(err error) { panic(err) },
+		})
+	if err != nil {
+		return nil, err
+	}
+	glcm := common.GetLifecycleMgr()
+	jsonKey, err = ioutil.ReadFile(glcm.GetEnvironmentVariable(common.EEnvironmentVariable.GoogleAppCredentials()))
+	if err != nil {
```

Comment:
It's already been gone over, but I'm also not comfortable with an unencrypted file on disk. What I'd suggest is that we have an alternate azcopy login where a user could insert their JSON via say, an environment variable or something for us to hold onto it. (or, just use the environment variable as the entire json file in the code)

Reply:
Could you elaborate a little more on what you mean?